server/src/controllers/ResourcesController.js

```diff
@@ -0,0 +1,170 @@
+   */
+  getResourceById: async (req, res) => {
+    try {
+      const resource = await Resource.findById(req.params.id);
+      res.status(200).json(resource);
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+
+  /**
+   * @route POST api/resource/add
+   * @description Add a new resource
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {String} req.body.topicId _id of the topic of this resource
+   * @param {String} req.body.title
+   * @param {String} req.body.description
+   * @param {String} req.body.link The link (URL) to the resource
+   */
+  addResource: async (req, res) => {
+    try {
+      const topic = await Topic.findById(req.body.topicId);
+      const resource = await Resource.create({
+        title: req.body.title,
+        description: req.body.description,
+        link: req.body.link,
+      });
+      topic.resources.push(resource._id);
+      topic.save();
+      res.status(201).json(resource);
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+  /**
+   * @route POST api/topic/add
+   * @description Add a new topic
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {String} req.body.name
+   */
+  addTopic: async (req, res) => {
+    console.log(req.body);
+    try {
+      const topic = await Topic.create({ name: req.body.name });
+      res.status(201).json(topic);
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+  /**
+   * @route PUT api/resource/edit/:id
+   * @description Update a resource
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {Object} req.body Update fields
+   * @param {String} req.params.id The _id of the resource
+   */
+  updateResourceById: async (req, res) => {
+    try {
+      const resource = await Resource.findByIdAndUpdate(
+        req.params.id,
+        req.body
+      );
+      res.status(201).json({ id: resource._id });
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+  /**
+   * @route PUT api/topic/edit/:id
+   * @description Update a topic
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {Object} req.body Update fields
+   * @param {String} req.params.id The _id of the topic
+   */
+  updateTopicById: async (req, res) => {
+    try {
+      const topic = await Topic.findByIdAndUpdate(req.params.id, req.body);
+      res.status(201).json({ id: topic._id });
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+  /**
+   * @route DELETE api/resource/delete/:id
+   * @description Delete a resource
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {String} req.params.id The _id of the resource
+   */
+  deleteResourceById: async (req, res) => {
+    try {
+      await Resource.findByIdAndDelete(req.params.id);
+      res.status(200).end();
```

Comment:
We have been using `res.status(204).json({})`. To maintain consistency, we should use that only :pray: